src/compiler/builderState.ts

```diff
@@ -121,26 +132,33 @@ namespace ts {
 
                         set.forEach(v => deleteFromMultimap(reverse, v, k));
                         forward.delete(k);
+                        version++;
                         return true;
                     },
                     set: (k, vSet) => {
-                        deleted?.delete(k);
+                        let changed = !!deleted?.delete(k);

```

Comment:
I'm pretty sure the value from `delete` is redundant because it would necessarily be accompanied by a multi-map update, but I saw no harm in consuming it.

Reply:
Since `Path` isn't an object, you need to explicitly check whether it was `undefined`, right?

Reply:
The initializer returns a boolean unless the `deleted` Set is `undefined`; the truthiness of a `Path` is never observed here.